### DISCUSSION TITLE
1 and 6 bit full adder

### DISCUSSION THREAD

**Discussion by Odie-123** [nicolacimmino/PLC-14500#14] [Show and tell]
Hi,
i have an example program for a 1-bit full adder for you which doesn't need a Scratch Register for intermediate results. I used this to do a 6-bit full adder (2nd example file).

Greetings,
Christof 
[1-bit-adder.asm.txt](https://github.com/nicolacimmino/PLC-14500/files/14894921/1-bit-adder.asm.txt)
[6-bit-adder.asm.txt](https://github.com/nicolacimmino/PLC-14500/files/14894922/6-bit-adder.asm.txt)

**Comment by nicolacimmino** [comment]
Thanks! Would you like to add them to the assembler/examples folder with a PR? I'd like to get some PRs.

If not just let me know and I will add them.

**Comment by Odie-123** [comment]
I am relatively new to GitHub (but i use a local git in my company). I guess i have to create an issue, then a branch for this issue (with the suggested files) and at the end a pull request. I created an issue but i can't create a branch - i guess i need write permissions for that. Or is there a different way, e.g. with forks?

**Reply by nicolacimmino** [reply]
yes, in this case you would need to fork the project, make changes and then request to create a PR.

Didn't think it would be too complicated. I cann add them myself and give you credit if that is ok with you.

**Comment by Odie-123** [comment]
No i can do it, it's probably a good exercise for me! ;-)

**Comment by Odie-123** [comment]
Done, see pull request.
